src/Functions/seriesPeriodDetect.cpp

```diff
@@ -0,0 +1,135 @@
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wshadow"
+#pragma clang diagnostic ignored "-Wextra-semi-stmt"
+#pragma clang diagnostic ignored "-Wzero-as-null-pointer-constant"
+#include "pocketfft_hdronly.h"
+#pragma clang diagnostic pop
+
+#include <cmath>
+#include <Columns/ColumnArray.h>
+#include <Columns/ColumnsNumber.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/IFunction.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+extern const int BAD_ARGUMENTS;
+extern const int ILLEGAL_COLUMN;
+}
+
+/*Detect Period in time series data using FFT.
+ * FFT - Fast Fourier transform (https://en.wikipedia.org/wiki/Fast_Fourier_transform)
```

Comment:
Since this is about the algorithms itself and not details of the C++ implementation, I suppose we should have that also in the user-facing documentation (markdown file).